store/generated_key/memstore/ephemeral_db/ephemeral_db.go

```diff
@@ -0,0 +1,123 @@
+package ephemeral_db
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/Layr-Labs/eigenda-proxy/common"
+	"github.com/Layr-Labs/eigenda-proxy/store/generated_key/memstore/memconfig"
+	"github.com/Layr-Labs/eigenda/api"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+)
+
+const (
+	DefaultPruneInterval = 500 * time.Millisecond
+)
+
+// DB ... An ephemeral && simple in-memory database used to emulate
+// an EigenDA network for dispersal/retrieval operations.
+type DB struct {
+	// knobs used to express artificial conditions for testing
+	config *memconfig.SafeConfig
+	log    logging.Logger
+
+	mu        sync.RWMutex
+	keyStarts map[string]time.Time // used for managing expiration
+	store     map[string][]byte    // db
+}
+
+// New ... constructor
+func New(ctx context.Context, cfg *memconfig.SafeConfig, log logging.Logger) *DB {
+	db := &DB{
+		config:    cfg,
+		keyStarts: make(map[string]time.Time),
+		store:     make(map[string][]byte),
+		log:       log,
+	}
+
+	if cfg.BlobExpiration() != 0 {
+		db.log.Info("ephemeral db expiration enabled for payload entries.", "time", cfg.BlobExpiration)
```

Comment:
just realized we don't document this behavior (setting BlobExpiration to 0 just means no expiration) in either the config or the flag. can you add it?

Reply:
[27bc9f6](https://github.com/Layr-Labs/eigenda-proxy/pull/305/commits/27bc9f6174c1498c61cc79b963ec52cb85b5ee88)